core/pkg/ingress/controller/controller.go

```diff
@@ -944,9 +956,6 @@ func (ic *GenericController) createServers(data []interface{},
 						servers[host].SSLCertificate = cert.PemFileName
 						servers[host].SSLPemChecksum = cert.PemSHA
 					}
-				} else {
-					servers[host].SSLCertificate = defaultPemFileName
-					servers[host].SSLPemChecksum = defaultPemSHA
 				}
```

Comment:
@bprashanth PR #459 adds `if tlsSecretName == ""` before looking for a secret. Could this be changed to "use the default secret" instead a warning? This would fix a misbehavior on haproxy ingress. Note that this change doesn't assign the default cert on a misconfigured ingress.